src/scheduler/Scheduler.cpp

```diff
@@ -342,11 +358,24 @@ faabric::util::SchedulingDecision Scheduler::makeSchedulingDecision(
         // At this point there's no more capacity in the system, so we
         // just need to overload locally
         if (remainder > 0) {
-            SPDLOG_DEBUG(
-              "Overloading {}/{} {} locally", remainder, nMessages, funcStr);
+            std::string overloadedHost = thisHost;
+
+            if (topologyHint ==
+                  faabric::util::SchedulingTopologyHint::NEVER_ALONE &&
+                hosts.size() > 0) {
```

Comment:
I think this would produce a clang-tidy warning about using `empty` instead, i.e. `!hosts.empty()` rather than `hosts.size() > 0`.